scripts/release-notes.py

```diff
@@ -0,0 +1,139 @@
+    commits_url = f"{base_url}/commits"
+    skipped_dependabot = 0
+
+    # If num_commits isn't set, get the number of commits made since the previous release tag.
+    if not num_commits:
+        num_commits = num_commits_since_prev_tag(token, base_url)
+
+    if not num_commits:
+        return
+
+    # Load commits
+    data = urllib.parse.urlencode({'per_page': num_commits})
+    req = Request(commits_url + '?' + data)
+    print(req.full_url)
+    req.add_header('Authorization', f'token {token}')
+    commits = json.loads(urlopen(req).read())
+    print('Retrieved', len(commits), 'commits')
+
+    # Load PR for each commit and print summary
+    for commit in commits:
+        sha = commit['sha']
+        author = commit['author']['login']
+
+        if exclude_dependabot and author == "dependabot[bot]":
+            skipped_dependabot += 1
+            continue
+
+        author_url = commit['author']['html_url']
+        msg_lines = commit['commit']['message'].split('\n')
+        msg = msg_lines[0].capitalize()
+        req = Request(f"{commits_url}/{sha}/pulls")
+        req.add_header('accept', accept_header)
+        req.add_header('Authorization', f'token {token}')
+        pulls = json.loads(urlopen(req).read())
+        if len(pulls) > 1:
+            print(f"WARNING: More than one pull request for commit {sha}")
+
+        # Handle commits without pull requests.
+        if not pulls:
+            short_sha = sha[:7]
+            commit_url = commit['html_url']
+            print(f'{point_md} {msg} ([@{author}]({author_url}) in [{short_sha}]({commit_url}))')
+            continue
+
+        pull = pulls[0]
+        pull_id = pull['number']
+        pull_url = pull['html_url']
+        msg = msg.replace(f'(#{pull_id})', '').strip()
+        print(f'{point_md} {msg} ([@{author}]({author_url}) in [#{pull_id}]({pull_url}))')
+
+    if skipped_dependabot:
+        print(f"\n(Skipped {skipped_dependabot} dependabot commit{'' if skipped_dependabot == 1 else 's'})")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='List changes based on git log for release notes.')
+
+    parser.add_argument('--token-file', type=str, default="~/.github_token",
+                        help='The file containing your personal github token to access the github API. ' +
+                             '(default: ~/.github_token)')
+    parser.add_argument('--repo', type=str, default='jaeger',
+                        help='The repository name to fetch commit logs from. (default: jaeger)')
+    parser.add_argument('--exclude-dependabot', action='store_true',
+                        help='Excludes dependabot commits. (default: false)')
+    parser.add_argument('--num-commits', type=int,
+                        help='Print this number of commits from git log. ' +
+                             '(default: number of commits before the previous tag)')
+    parser.add_argument('--point-markdown-char', type=str, default='*',
```

Comment:
This script is essentially a copy-paste of https://github.com/jaegertracing/jaeger/blob/main/scripts/release-notes.py with some minor enhancements such as making the point markdown char configurable for consistency with the existing convention, as well as auto-capitalizing sentences in each point, which have been used to generate the above change log.

If we think these improvements are worthwhile keeping, I can sync it up with the `jaeger` repo.

Reply:
I would actually prefer to have just a single copy of the full script in the main repo, and in this repo have a thin wrapper that will download the other script from github.

Reply:
Do you think we should do the same for `scripts/draft-release.py` as well? We'd need to make some minor differences  that would need to be either make configurable or opinionated.